sentence_transformers/losses/MatryoshkaLoss.py

```diff
@@ -110,23 +110,39 @@ def __init__(
         self.matryoshka_weights = matryoshka_weights
         self.n_dims_per_step = n_dims_per_step
 
+    @property
```

Comment:
Using `property` and `setter` so that any attempt to set the attribute, whether in the init or by the user, triggers the input check. Lmk if you prefer to just do it in the init or forward methods.

Reply:
Maybe a better idea is to put the check in `ForwardDecorator.shrink`? It's simpler and will catch more errors